### PR TITLE
Parse command line options using click

### PR DESCRIPTION
**Pull request by stefsmeets**
Hi @v1kko, I was playing with `click` today to see how we could use it instead of `argparse`. This is what I came up with. I quite like the result, what do you think?

### CODE REVIEW

File: duqtools/status.py [hunk 1]
```diff
@@ -180,11 +180,12 @@ def detailed_status(self):
...
 
 def status(progress: bool, detailed: bool, **kwargs):
-    tracker = Status()
-
-    if detailed:
-        tracker.detailed_status()
-    elif progress:
-        tracker.progress_status()
-    else:
-        tracker.simple_status()
```
Comment (v1kko) [inline duqtools/status.py:190]: I think you overwrote your own previous adjustements here? I think the logic is still needed right?

File: duqtools/status.py [hunk 2]
```diff
@@ -180,11 +180,12 @@ def detailed_status(self):
...
+    """Show status of runs.
+
+    Parameters
+    ----------
+    progress : bool
+        Show progress bar.
+    detailed : bool
+        Show detailed progress for every job.
+    """
```
Comment (v1kko) [inline duqtools/status.py:191]:
```suggestion
    """

    tracker = Status()

    if detailed:
        tracker.detailed_status()
    elif progress:
        tracker.progress_status()
    else:
        tracker.simple_status()
```

**Review by v1kko — CHANGES_REQUESTED** [review]
Great changes, I guess functionally it works the same, and we get a bonus python api

**Comment by stefsmeets** [comment]
Thanks! Yeah, functionally it's pretty much the same, except that the order of the options is more strictly enforced by `click` (we sort of worked around it with argparse, but it doesn't like it either).

Meaning you must do `duqtools -c config.yaml --debug create` instead of `duqtools create -c config.yaml --debug`.

**Pull request merged**